File: src/ast/QualifiedName.java
package ast;

import java.io.StringWriter;

public class QualifiedName extends BaseNode {
    private BaseNode _qualifier;
    private BaseNode _name;

    public QualifiedName(BaseNode qualifier, BaseNode name) {
        super(NodeType.QualifiedName);
        _qualifier = qualifier;
        _name = name;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        _qualifier.Print(writer);
        writer.write("::");
        _name.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/QualifiedName.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class QualifiedName(private val qualifier: BaseNode?, val name: BaseNode) : BaseNode(NodeType.QualifiedName) {
    public override fun printLeft(writer: StringWriter) {
        qualifier!!.print(writer)
        writer.write("::")
        name.print(writer)
    }

}